Дипломный проект. Задание 3: веб-приложение

loysopondohva opened the pull request:
Автотесты для проверки веб-приложения Stellar Burgers.

#### Проверка основного функционала (`test_main_page.py`)
- Проверка перехода по клику на «Конструктор»;
- Проверка перехода по клику на «Лента заказов»;
- Проверка если кликнуть на ингредиент, появится всплывающее окно с деталями;
- Всплывающее окно закрывается кликом по крестику;
- при добавлении ингредиента в заказ счётчик этого ингредиента увеличивается.
залогиненный пользователь может оформить заказ.

#### Раздел «Лента заказов» (`test_order_feed_page.py`)
- Проверка при создании нового заказа счётчик «Выполнено за всё время» увеличивается;
- Проверка при создании нового заказа счётчик «Выполнено за сегодня» увеличивается;
- Проверка после оформления заказа его номер появляется в разделе «В работе».

loysopondohva committed:
Finish

loysopondohva committed:
requirements

loysopondohva committed:
remove imports

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,31 @@
+import pytest  # Обязательно импортируем pytest
+from selenium import webdriver
+
+class WebdriverFactory:
```
Нужно исправить: модуль conftest предназначен только для фикстур и хуков.

loysopondohva replied:
Исправлено, класс вынесен в файл ./helpers/webdrivers.py

ShamilovR commented on pages/main_page.py:
```diff
@@ -0,0 +1,82 @@
+
+    @allure.step('Проверить, отображается всплывающее окно с информацией об ингредиенте')
+    def is_ingredients_details_displayed(self):
+        element = self.driver.find_element(*MainPageLocators.CLOSE_INGREDIENT_DETAILS_BUTTON)
```
Нужно исправить: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

loysopondohva replied:
Исправлено, переопределил через метод BasePage

ShamilovR commented on tests/test_order_feed_page.py:
```diff
@@ -0,0 +1,124 @@
+            user_password = data.Credentials.password
+            account_page.login_to_account(user_email, user_password)
+
+        order_feed_page.click_to_order_feed_button()
```
Нужно исправить: тесовый сценарий должен заканчиваться проверкой (assert`ом)

loysopondohva replied:
Это случайно осталось от проверок FireFox и перекрывающих окон

loysopondohva committed:
Changes after review:
- coftest.py Class WebdriverFactory moved to helpers/webdriver.py
- main_page.py: changes method is_ingredients_details_displayed()
- test_order_feed_page.py: remove test transition
